From bb9bb7286a4c345442946dc2ce3c9e7f67e96d4d Mon Sep 17 00:00:00 2001
From: Steve Dickson <steved@redhat.com>
Date: Wed, 27 May 2026 11:42:11 -0400
Subject: [PATCH] rpcinfo: stack buffer overflow in rpcinfo rpcbaddrlist()

rpcinfo's rpcbaddrlist() formats two server-controlled, unbounded XDR strings into a fixed 128-byte stack buffer with sprintf(). A malicious or on-path rpcbind server overflows it when a user runs:
rpcinfo -l <host> <prognum> <versnum>

Reported-by: Michalis Vasileiadis <vmihalis.tmd@gmail.com>
Signed-off-by: Steve Dickson <steved@redhat.com>
---
 src/rpcinfo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rpcinfo.c b/src/rpcinfo.c
index c59e8b4..30d535e 100644
--- a/src/rpcinfo.c
+++ b/src/rpcinfo.c
@@ -1104,7 +1104,7 @@ rpcbaddrlist (
 
 	  re = &head->rpcb_entry_map;
 	  printf ("%10u%3u    ", parms.r_prog, parms.r_vers);
-	  sprintf (buf, "%s/%s/%s ",
+	  snprintf (buf, sizeof(buf), "%s/%s/%s ",
 		   re->r_nc_protofmly, re->r_nc_proto,
 		   re->r_nc_semantics == NC_TPI_CLTS ? "clts" :
 		   re->r_nc_semantics == NC_TPI_COTS ? "cots" : "cots_ord");
-- 
2.54.0

